DEVELOPERS.md

```diff
@@ -0,0 +1,159 @@
+# Waku Objects Developers Documentation
+
+## Table of Contents
+
+- [Overview](#overview)
+  - [Terminology](#terminology)
+- [Waku Objects Chat Protocol](#waku-objects-chat-protocol)
+
+  - [Message types](#message-types)
+    - [UserMessage type](#usermessage-type)
+    - [InviteMessage type](#invitemessage-type)
+    - [DataMessage type](#datamessage-type)
+  - [Private chat](#private-chat)
+  - [Group chat](#group-chat)
+  - [Other data definitions](#other-data-definitions)
+
+- Waku Object SDK
+  - Overview
+  - Types
+  - External objects
+  - Creating your own object type
+  - Existing implementations
+
+## Overview
+
+Waku Objects is an open, modular system for transactional mini-applications to be built inside a chat application. Open, because it uses open standards for interoperability and provides a default [web based implementation](https://www.wakuplay.im/). Modular, because it may be implemented in any chat application that supports structured data messages that are sent alongside user messages. Transactional, because it supports blockchain transactions and smart contract interactions on Ethereum compatible blockchains.
+
+The Waku Object concept enables building social applications, where the social context is provided by the chat (private 1on1 or group at the moment) so that there is already some implied trust between the participants. The Waku Object SDK provides an interface for objects to communicate easily with the other participants' objects and offer safe services without leaking private or personal data.
```

Comment:
> ...and offer safe services without leaking private or personal data

I am not sure we can claim this right now.

Reply:
I reworded it to sound more aspirational than factual. 

What I meant to say here is that the Waku Object interface is designed to share the least information possible in a sandboxed environment so that the possibility of leaking private or personal data is minimised. We don't share messages, we don't share the private key, the only thing we share currently is the name, avatars and addresses of the participant. From these the names and avatars are arbitrary, so it's only the addresses that can be considered PII. We can even use generated addresses and replace them in the application so that the objects has not even knowledge of the real addresses. I will open an issue about this :smile:

Reply:
https://github.com/logos-innovation-lab/waku-objects-playground/issues/448